test_server.py

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python
+
+import pytest
+import socket
+from http_server import response_ok, response_error
+
+
+@pytest.fixture()
+def make_client():
+    ADDR = ("127.0.0.1", 8010)
+    client = socket.socket(
+        socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_IP
+    )
+    client.connect(ADDR)
+    return client
+
+
+def helper(client, msg):
+    try:
+        client.sendall(msg)
+        client.shutdown(socket.SHUT_WR)
+        response = ""
+        while True:
+            part = client.recv(16)
+            response += part
+            if len(part) < 16:
+                client.close()
+                return response
+    except Exception as e:
+        print e
```

Comment:
Good helper method.  Only thing I'd change is to allow exceptions to bubble up instead of printing and silencing them.